handlers/identity-backfill/src/main/scala/com/gu/identity/GetByEmail.scala

```diff
@@ -3,18 +3,17 @@ package com.gu.identity
 import com.gu.identity.GetByEmail.RawWireModel.{User, UserResponse}
 import com.gu.identityBackfill.Types.EmailAddress
 import com.gu.identityBackfill.salesforce.UpdateSalesforceIdentityId.IdentityId
-import com.gu.util.config.ConfigLocation
-import okhttp3.{HttpUrl, Request, Response}
-import play.api.libs.json.{Json, Reads}
-import scalaz.syntax.std.either._
-import scalaz.{-\/, \/, \/-}
+import com.gu.util.resthttp.HttpOp.HttpOpWrapper
+import com.gu.util.resthttp.RestRequestMaker
+import com.gu.util.resthttp.RestRequestMaker.{GetRequestWithParams, RelativePath, UrlParams}
+import com.gu.util.resthttp.Types.{ClientFailableOp, ClientSuccess, GenericError}
+import play.api.libs.json.{JsValue, Json, Reads}
 
 object GetByEmail {
 
-  sealed trait ApiError
-  case class OtherError(message: String) extends ApiError
-  case object NotFound extends ApiError
-  case object NotValidated extends ApiError
+  sealed trait MaybeValidatedEmail
```

Comment:
Not sure how much better this is, but we could just have something like:

```suggestion
  sealed trait IdentityEmailAddress
  case class ValidatedEmailAddress(identityId: IdentityId) extends IdentityEmailAddress
  case object UnvalidatedEmailAddress extends IdentityEmailAddress
```

Reply:
I think yours is similarly confusing, I think this is better:
```scala
  sealed trait IdentityAccount
  case class IdentityAccountWithValidatedEmail(identityId: IdentityId) extends IdentityAccount
  case object IdentityAccountWithUnvalidatedEmail extends IdentityAccount
```